docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+- Bubble up more information about both CP and worker Machines, ensuring consistent way across Cluster API resources.
+    - Standardize replica counters and bubble them up to the Cluster resource.
+    - Standardize control plane, MachineDeployment, Machine pool availability, and bubble them up to the Cluster resource.
+- Introduce missing signals about connectivity to workload clusters, thus enabling to mark all the conditions
+  depending on such connectivity being working with status Unknown after a certain amount of time.
+- Introduce a cleaner signal about Cluster API resources lifecycle transitions, e.g. scaling up or updating.
+- Ensure everything in status can be used as a signal informing monitoring tools/automation on top of Cluster 
+  about lifecycle transitions/state of the Cluster and the underlying components as well.
+
+### Non-Goals/Future Work
+
+- Resolving all the idiosyncrasies that exists in Cluster API, core Kubernetes, the rest of the ecosystem.
+  (Let’s stay focused in Cluster API and keep improving incrementally).
+
+## Proposal
+
+This proposal groups a set of changes to status fields in Cluster API resources.
+
+Some of those changes could be considered straight forward, e.g.
+
+- K8s API conventions suggest to deprecate and remove `phase` fields from status, Cluster API is going to align to this recommendation
+  (and improve Conditions to provide similar or even a better info as a replacement).
+- K8s resources do not have a concept similar to "terminal feature" existing in Cluster API resources, and users approaching
+  the project are struggling with this idea; in some cases also provider's implementers are struggling with it.
+  Accordingly, Cluster API resources are dropping `FailureReason` and `FailureMessage` fields (terminal failures should be surfaced using
+  conditions, like any other error/warning/message)
+- Bubble up more information about both CP and worker Machines to the Cluster level.
+
+Some other changes requires a little bit more context, which is provided in following paragraphs:
+
+- Review and standardize the usage of the concept of readiness and availability to align to K8s API conventions / 
+  conditions used in core K8s objects like `Pod`, `Node`, `Deployment`, `ReplicaSet` etc. 
+- Transition to K8s API conventions fully aligned conditions types/condition management (and thus deprecation of
+  the Cluster API "custom" guidelines for conditions).
+
+The last set of changes is a consequence of the above changes, or small improvements to address feedback received
+over time; changes in this group will be detailed case by case in the following paragraphs, a few examples:
+
+- Change the semantic of ReadyReplica counters to use Machine's Ready condition instead of Node's Ready condition.
+  (so everywhere Ready is used for a Machine it always means the same thing)
+- Add a new condition monitoring the status of the connectivity to workload clusters (`RemoteConnectionProbe`).
```

Comment:
Amazing

Reply:
Thanks for the feedback, I agree this is a very welcome improvement! 
Resolving this to keep the number of open threads under control 😅